src/main/java/software/amazon/awssdk/crt/s3/S3Client.java

```diff
@@ -122,6 +122,12 @@ private void onShutdownComplete() {
 
     public S3MetaRequest makeMetaRequest(S3MetaRequestOptions options) {
 
+        if(isNull()) {
```

Comment:
The S3Client is intended to be a "thread safe" class, but I don't think there's any synchronization going on between `makeMetaRequest()` and `close()`

Reply:
Looking at the other places, this is a common problem that we don't have any synchronization for close function. 

- https://github.com/awslabs/aws-crt-java/blob/a8bec8665c547eaadcc00388a6e2aab1faa3eb1e/src/main/java/software/amazon/awssdk/crt/http/HttpClientConnection.java#L44
- https://github.com/awslabs/aws-crt-java/blob/a8bec8665c547eaadcc00388a6e2aab1faa3eb1e/src/main/java/software/amazon/awssdk/crt/http/Http2ClientConnection.java#L187-L187
- https://github.com/awslabs/aws-crt-java/blob/a8bec8665c547eaadcc00388a6e2aab1faa3eb1e/src/main/java/software/amazon/awssdk/crt/eventstream/ClientConnection.java#L109
- https://github.com/awslabs/aws-crt-java/blob/a8bec8665c547eaadcc00388a6e2aab1faa3eb1e/src/main/java/software/amazon/awssdk/crt/http/HttpStreamBase.java#L97
- https://github.com/awslabs/aws-crt-java/blob/e50af06926946768aaa1e847b38298990280e3db/src/main/java/software/amazon/awssdk/crt/io/TlsKeyOperation.java#L139

We can solve this problem when we encounter it.